kernel: bump 6.6 to 6.6.65

graysky2 opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.65

Update to checksum only/no patches required a rebase. Per the changelog, this is a trivial upstream bump with only 4 commits, 2 of which being reverts.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne
Run-tested: x86/64/AMD Cezanne

graysky2 commented:
I need someone to review `bcm27xx/patches-6.6/950-0829-vc4-hvs-Add-support-for-D0-register-changes.patch`

namiltd commented on target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch:
```diff
@@ -10,22 +10,23 @@ Subject: [PATCH] net/usb/qmi_wwan: add MeigLink modem support
  		.driver_info = (unsigned long)&qmi_wwan_info,
  	},
-+	{	/* Meiglink SGM828 */
++	{       /* Meiglink SGM828 */
```
The rest of the code is TAB

graysky2 replied:
fixed

namiltd commented on target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch:
```diff
@@ -10,22 +10,23 @@ Subject: [PATCH] net/usb/qmi_wwan: add MeigLink modem support
  	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0125)},	/* Quectel EC25, EC20 R2.0  Mini PCIe */
  	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0306)},	/* Quectel EP06/EG06/EM06 */
- 	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0512)},	/* Quectel EG12/EM12 */
+@@ -1091,6 +1096,7 @@ static const struct usb_device_id produc
```
The rest of the code is TAB

graysky2 replied:
fixed

namiltd commented on target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch:
```diff
@@ -10,22 +10,23 @@ Subject: [PATCH] net/usb/qmi_wwan: add MeigLink modem support
  	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0800)},	/* Quectel RM500Q-GL */
  	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0801)},	/* Quectel RM520N */
-+	{QMI_MATCH_FF_FF_FF(0x05c6, 0xf601)},	/* MeigLink SLM750 */
++	{QMI_MATCH_FF_FF_FF(0x05c6, 0xf601)},   /* MeigLink SLM750 */
```
The rest of the code is TAB

graysky2 replied:
fixed

Beginner-Go commented:
`bcm27xx/patches-6.6/950-0597-drm-vc4-hdmi-Avoid-hang-with-debug-registers-when-su.patch` Delete items that are not written to the update log

namiltd commented:
In the file 950-0007-drm-vc4-Add-firmware-kms-mode.patch
replace
```
(vc4->is_vc5)
```
with
```
(vc4->vc4_gen == VC4_GEN_5)
```

graysky2 commented:
@namiltd - For the `:%s/(vc4->is_vc5)/(vc4->vc4_gen == VC4_GEN_5)/g` I assume all occurances?

graysky2 commented on target/linux/bcm27xx/patches-6.6/950-0007-drm-vc4-Add-firmware-kms-mode.patch:
```diff
@@ -2329,7 +2329,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  
--	if (vc4->is_vc5) {
+-	if (vc4->vc4_gen == VC4_GEN_5) {
 +	if (vc4->is_vc5 && !vc4->firmware_kms) {
```
Got it

namiltd commented:
Revert to previous 950-0007-drm-vc4-Add-firmware-kms-mode.patch
and add to 950-0585-drm-vc4-Introduce-generation-number-enum.patch add a patch changing in file vc4_hvs.c:
```
@@ -224,6 +224,6 @@
 	if (!drm_dev_enter(drm, &idx))
 		return;
 
-	if (hvs->vc4->is_vc5)
+	if (hvs->vc4->vc4_gen == VC4_GEN_5)  
 		return;
 
 	/* The LUT memory is laid out with each HVS channel in order,
```

graysky2 commented:
@namiltd - sorry, I am not sure what you're suggesting.  Can you send a commit?

namiltd commented:
[950-0007-drm-vc4-Add-firmware-kms-mode.patch](https://github.com/user-attachments/files/18068596/950-0007-drm-vc4-Add-firmware-kms-mode.patch)
I have corrected it with a text editor, so I apologize for any shortcomings.

graysky2 committed:
kernel: bump 6.6 to 6.6.64

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.64

Manually rebased:
	generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch
	bcm27xx/patches-6.6/950-0585-drm-vc4-Introduce-generation-number-enum.patch
	bcm27xx/patches-6.6/950-0610-drm-vc4-hvs-Support-BCM2712-HVS.patch
	bcm27xx/patches-6.6/950-0829-vc4-hvs-Add-support-for-D0-register-changes.patch

Removed upstreamed:
	bcm27xx/patches-6.6/950-0597-drm-vc4-hdmi-Avoid-hang-with-debug-registers-when-su.patch[1]
	bcm27xx/patches-6.6/950-0599-drm-vc4-Fix-dlist-debug-not-resetting-the-next-entry.patch[2]
	bcm27xx/patches-6.6/950-0600-drm-vc4-Remove-incorrect-limit-from-hvs_dlist-debugf.patch[3]
	bcm27xx/patches-6.6/950-0708-drm-vc4-Correct-logic-on-stopping-an-HVS-channel.patch[4]
	ramips/patches-6.6/002-01-v6.13-clk-ralink-mtmips-fix-clock-plan-for-Ralink-SoC-RT38.patch[5]
	ramips/patches-6.6/002-02-v6.13-clk-ralink-mtmips-fix-clocks-probe-order-in-oldest-r.patch[6]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.64&id=223ee2567a55e4f80315c768d2969e6a3b9fb23d
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.64&id=8182b5ca19c6f173b6498d1c6d3e4b034b76bbde
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.64&id=52c1716f65a558174e381360bd88f18dae4be85c
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.64&id=9728b508b01a5eeeac79ceb676364c674dd951ac
5. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.64&id=f85a1d06afbcc57ac44176db8f9d7a934979952c
6. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.64&id=fbb13732c6ffa9d58cedafabcd5ce8fd7ef8ae5a

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/17217
Signed-off-by: Robert Marko <robimarko@gmail.com>

rmandrad commented:
tested pull from @namiltd successfully on the mediatek/filogic bananapi4

namiltd commented:
Here you go: https://github.com/graysky2/openwrt/pull/4

zxlhhyccc commented:
````
HDRINST usr/include/asm/poll.h
  INSTALL /__w/openwrt/openwrt/openwrt/build_dir/target-arm-openwrt-linux-muslgnueabi_musl/linux-bcm27xx_bcm2708/linux-6.6.64/user_headers/include
make[4]: Leaving directory '/__w/openwrt/openwrt/openwrt/build_dir/target-arm-openwrt-linux-muslgnueabi_musl/linux-bcm[27](https://github.com/openwrt/openwrt/actions/runs/12241818685/job/34147907735?pr=17217#step:31:28)xx_bcm2708/linux-6.6.64'
drivers/gpu/drm/vc4/vc4_hvs.c: In function 'vc4_hvs_lut_load':
drivers/gpu/drm/vc4/vc4_hvs.c:524:21: error: 'struct vc4_dev' has no member named 'is_vc5'
  524 |         if (hvs->vc4->is_vc5)
      |                     ^~
make[10]: *** [scripts/Makefile.build:243: drivers/gpu/drm/vc4/vc4_hvs.o] Error 1
make[9]: *** [scripts/Makefile.build:480: drivers/gpu/drm/vc4] Error 2
make[8]: *** [scripts/Makefile.build:480: drivers/gpu/drm] Error 2
make[7]: *** [scripts/Makefile.build:480: drivers/gpu] Error 2
make[6]: *** [scripts/Makefile.build:480: drivers] Error 2
make[6]: *** Waiting for unfinished jobs....
````

namiltd commented:
@zxlhhyccc Fixed in https://github.com/graysky2/openwrt/pull/4.

namiltd commented:
Download https://github.com/graysky2/openwrt/pull/4.patch
git reset --hard ff5f77ecdd218a9451bd6c531be86c6335fadd7a
git am location_of_file-4.patch
git push --force

namiltd commented:
git commit --amend

and change:

Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: John Audia <therealgraysky@proton.me>

git push --force

namiltd commented:
From:
```
Signed-off-by: John Audia <therealgraysky@proton.me>
Co-authoered-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
```
to
```
Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: John Audia <therealgraysky@proton.me>
``
And update the PR description (different than in Commit).

namiltd commented:
Well, because I am as an author.
The easiest way:
- copy files somewhere
- rollback to before patch
- ecover files back
 - confirm and enter a comment

namiltd commented:
.

namiltd commented:
Build system: x86/64
Build-tested: ramips/tplink_archer-c5-v4
Run-tested: ramips/tplink_archer-c5-v4

hauke commented:
You can change the author of a commit with `git commit --amend --author="John Audia <therealgraysky@proton.me>"`

namiltd commented:
Errors are due tohttps://github.com/openwrt/openwrt/pull/16397#issuecomment-2538723118
You can make a quick patch.

graysky2 committed:
kernel: bump 6.6 to 6.6.65

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.65

Update to checksum only/no patches required a rebase.
Per the changelog, this is a trivial upstream bump with only 4
commits, 2 of which being reverts.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne
Run-tested: x86/64/AMD Cezanne

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/17217
Signed-off-by: Robert Marko <robimarko@gmail.com>

namiltd commented:
Please clean up

```
Signed-off-by: John Audia <therealgraysky@proton.me>
Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
```
to
```
Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: John Audia <therealgraysky@proton.me>
```
so that this PR can be approved

robimarko commented:
Thanks! Rebased on top of main and merged!
